File: win-prog/ProgWinCS/Images and Bitmaps/PartialImage/PartialImage.cs
//-------------------------------------------
// PartialImage.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class PartialImage: PrintableForm
{
     Image image;

     public new static void Main()
     {
          Application.Run(new PartialImage());
     }
     public PartialImage()
     {
          Text = "Partial Image";

          image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Rectangle rect = new Rectangle(95, 5, 50, 55);

          grfx.DrawImage(image, 0, 0, rect, GraphicsUnit.Pixel);
     }
}

File: win-prog/ProgWinVB/Images and Bitmaps/PartialImage/PartialImage.vb
'---------------------------------------------
' PartialImage.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class PartialImage
    Inherits PrintableForm

    Private img As Image

    Shared Shadows Sub Main()
        Application.Run(new PartialImage())
    End Sub

    Sub New()
        Text = "Partial Image"
        img = Image.FromFile("..\..\Apollo11FullColor.jpg")
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim rect As New Rectangle(95, 5, 50, 55)
        grfx.DrawImage(img, 0, 0, rect, GraphicsUnit.Pixel)
    End Sub
End Class
